clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -487,7 +487,12 @@ else if (strategy == OffsetResetStrategy.LATEST)
                 } else {
                     TopicPartition partition = nextInLineRecords.partition;
                     fetchedPartition = partition;
-                    fetchedOffsets = subscriptions.position(partition);
+                    if (subscriptions.isAssigned(partition)) {
```

Comment:
so @becketqin to confirm, you are saying @hachikuji's suggestion on `fetchedOffsets = nextInLineRecords.fetchOffset;` is the way to go? 
I ask because it takes 7 hours to run system tests :)